src/salaryAndBenefits/utils/calculateSalary.ts

```diff
@@ -0,0 +1,30 @@
+import { payscale } from "./salaryData";
+  
+interface PayScale {
+    [year: string]: {
+      [examinationYear: string]: string;
+    };
+  }
+  
+  const salaryPayscale: PayScale = payscale;
+  
+  export function calculateSalary(
+    currentYear: number,
+    examinationYear: number,
+    degree: string
+  ): string {
+
+    if (examinationYear >= currentYear) {
+        //TODO: add error handling for data that is not in the payscale
+        console.log("Examination year cannot be in the future.");
+    }      
+
+    const adjustedYear = (examinationYear + (degree === "Bachelor" ? 1 : 0)).toString();
+    return salaryPayscale[currentYear][adjustedYear];
```

Comment:
Shouldn't `currentYear` also be converted to a string?

Reply:
JavaScript handles indexes as string internally anyway, so it will run regarless 🙃

Reply:
Does typescript do this as well?

Reply:
No difference at runtime, so yes. But in theory some checks might complain about the type difference.